app/frontend/javascript/survey_widget.js

```diff
@@ -0,0 +1,88 @@
+// This is a fairly brute-force
+// attempt to make sure that if
+// you don't want to fill in the survey
+// or already did fill it in, you
+// won't be asked again for a long time.
+
+// Despite our best efforts,
+// the survey was not registering
+// the fact that users had dismissed it
+// or already filled it in.
+
+// Note that SurveySparrow has its own
+// mechanisms in place to attempt to ensure
+// that users only fill out the survey once;
+// we just don't trust them.
+jQuery( document ).ready(function() {
+  if (jQuery('#ss_survey_widget').length != 1) {
+    return;
+  }
+  if (surveyAlreadyShown()) {
+    window.console.log("You've already clicked on the survey! Not gonna show it.")
+  }
+  else {
+    window.console.log("Launching the survey.")
+    sparrowLaunch({
+      //add custom params here
+    })
+  }
+  // Any click on the main survey div is
+  // interpreted as "user has started survey."
+  // Once they start the survey, never show
+  // this user the survey again for an entire year.
+  jQuery ('#ss_survey_widget').click(userHasStartedSurvey);
+});
+
+
+function surveyAlreadyShown() {
+  return document.cookie.match(/surveyAlreadyShown=true/) != null;
+}
+
+function userHasStartedSurvey(event) {
+  event.preventDefault();
+  var expiratonStr = new Date(new Date()
+    .setFullYear(new Date()
+    .getFullYear() + 3))
+    .toString();
+  document.cookie = "surveyAlreadyShown=true; path=/; expires=" + expiratonStr;
+  window.console.log("Set the cookie!");
+}
+
+// This is taken from the survey sparrow site.
```

Comment:
Can you provide a URL, where did you get this from?  Among other things, if it *changes* in the future from SurveySparrow's end, we want to be able to track down where to get a new version. 

Is this actually a supported thing to do, by SurveySparrow , to copy this code from them? Or an unsupported hack? If the latter, I don't love it... there's no supported by SurveySparrow way to do what we need? Can we think about if there is a different way to do this supported by SurveySparrow? If not, we should definitely file an issue with them telling them what we want to do; and there should also be comments here explaining what we're doing.

Reply:
Generally linking to relevant Survey Sparrow docs URL in a comment _somewhere_ in this PR  is also probably a good idea. As well as putting it in the Github PR descirption itself; if you disappear and someone who isn't you (or me!) has to maintain this, leave them cookie crumbs to figure out what they're doing.

Reply:
Definitely. Edited the PR and put in some basic notes in the code as well.